experiment/build/generate_cloudbuild.py

```diff
@@ -13,65 +13,119 @@
 # limitations under the License.
 """Generates Cloud Build specification"""
 
-import argparse
-import json
-import yaml
-
-from experiment.build import docker_images
-
-
-# TODO(Tanq16): Add unit test for this.
-def create_cloud_build_spec(buildable_images, docker_registry):
-    """Returns Cloud Build specificatiion."""
-
-    cloud_build_spec = {}
-    cloud_build_spec['steps'] = []
-    cloud_build_spec['images'] = []
-
-    for name, image in buildable_images.items():
-        step = {}
-        step['id'] = name
-        step['name'] = 'gcr.io/cloud-builders/docker'
-        step['args'] = []
-        step['args'] += ['--tag', image['tag']]
-        step['args'] += ['--cache-from', docker_registry + image['tag']]
-        step['args'] += ['--build-arg', 'BUILDKIT_INLINE_CACHE=1']
-        if 'build_arg' in image:
-            for build_arg in image['build_arg']:
-                step['args'] += ['--build-arg', build_arg]
-        if 'dockerfile' in image:
-            step['args'] += ['--file', image['dockerfile']]
-        step['args'] += [image['context']]
-        if 'depends_on' in image:
-            step['wait_for'] = []
-            for dep in image['depends_on']:
-                step['wait_for'] += [dep]
-        cloud_build_spec['images'].append(name)
+import os
+import posixpath
+
+from common import yaml_utils
+from common import experiment_utils
+from common import experiment_path as exp_path
+from common.utils import ROOT_DIR
+from experiment.build import build_utils
+
+BASE_TAG = 'gcr.io/fuzzbench'
+
+
+def get_experiment_tag_for_image(image, experiment=True):
+    """Returns the registry with the experiment tag for given image."""
+    if not experiment:
+        return posixpath.join(BASE_TAG, image['tag']) + ':test-experiment'
+    tag = posixpath.join(experiment_utils.get_base_docker_tag(), image['tag'])
+    tag += ':' + experiment_utils.get_experiment_name()
+    return tag
+
+
+def coverage_steps(benchmark):
+    """Returns GCB run steps for coverage builds."""
+    coverage_binaries_dir = exp_path.filestore(
+        build_utils.get_coverage_binaries_dir())
+    steps = [{
+        'name':
+            'gcr.io/cloud-builders/docker',
+        'args': [
+            'run', '-v', '/workspace/out:/host-out',
+            posixpath.join(experiment_utils.get_base_docker_tag(), 'builders',
+                           'coverage', benchmark) + ':' +
+            experiment_utils.get_experiment_name(), '/bin/bash', '-c',
+            'cd /out; tar -czvf /host-out/coverage-build-' + benchmark +
+            '.tar.gz *'
+        ]
+    }]
+    step = {'name': 'gcr.io/cloud-builders/gsutil'}
+    step['args'] = [
+        '-m', 'cp', '/workspace/out/coverage-build-' + benchmark + '.tar.gz',
+        coverage_binaries_dir + '/'
+    ]
+    steps.append(step)
+    return steps
+
+
+def create_cloud_build_spec(image_templates,
+                            benchmark='',
+                            experiment=True,
```

Comment:
is_experiment is probably the better solution.
Frankly I'm not sure why we need this. Currently I fake the experiment when I want to use GCB outside of an experiment. I think that's acceptable.

Reply:
I used it to generate exactly with "gcr.io/fuzzbench" and "test-experiment", to create `docker/base-images.yaml`. Without it it would need theenv vars.

Reply:
You don't need to remove it now, but I'm fine with the env vars being required. This is what we do currently to run builder.py standalone right?

Reply:
Yes, right, so basically I'll assume whoever generates base-images.yaml will set the required env vars.